pico/types/src/main/java/io/helidon/pico/types/TypeName.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.types;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+/**
+ * TypeName is similar to {@link java.lang.reflect.Type} in its most basic use case. The {@link #getName()} returns the package +
+ * class name tuple for the given type (i.e., the canonical type name).
+ * <p>
+ *
+ * This class also provides a number of methods that are typically found in {@link java.lang.Class} that can be used to avoid
+ * classloading resolution:
+ * <li>{@link #getPackageName()} and {@link #getClassName()} - access to the package and simple class names.
+ * <li>{@link #isPrimitive()} and {@link #isArray()} - access to flags that is typically found in {@link java.lang.Class}.
+ * <p>
+ *
+ * Additionally, this class offers a number of additional methods that are useful for handling generics:
+ * <li>{@link #isGeneric()} - true when this type is declared to include generics (i.e., has type arguments).
+ * <li>{@link #isWildcard()} - true if using wildcard generics usage.
+ * <li>{@link #getTypeArguments()} - access to generics / parametrized type information.
+ * <p>
+ *
+ * Finally, this class offers a number of methods that are helpful for code generation:
+ * <li>{@link #getDeclaredName()} and {@link #getFQName()}.
+ */
+public interface TypeName extends Comparable<TypeName> {
+
+    /**
+     * Functions the same as {@link Class#getPackageName()}.
+     *
+     * @return the package name
+     */
+    @JsonIgnore
+    String getPackageName();
```

Comment:
Such an annotation would only make sense if you decide to send `TypeName` through Jackson. As that is not the case, the annotations are useless for target teams. 
This type should only be used by Helidon team (or team extending Pico).

Reply:
(Neutral on all this; just here to note the existence of [Jackson mixins](https://medium.com/@shankar.ganesh.1234/jackson-mixin-a-simple-guide-to-a-powerful-feature-d984341dc9e2), which is the normal way someone can "put" a Jackson annotation on a class that does not or cannot have Jackson annotations on it.)

Reply:
Chances are better than not that consumers of Helidon are using Jackson and Lombok.

And in such cases then we are implicitly hurting them by not including JsonIgnore.  I therefore would maintain my point that it doesn't harm anyone (internally or externally) by us having a provided/static module dependency here, but it can in fact hurt our customers that may be using it in these scenarios where they are using Jackson, Lombok, etc.

See https://medium.com/@mrkrijgsman/flexible-immutability-a475346d9518

Reply:
What about defining this field as `transient`?

Reply:
I agree, it's odd to use Jackson annotations here. But perhaps the PR is too narrow focused to understand the bigger picture.

Reply:
i took it out of the contracts for this PR, but we will see these again when we get to the impl modules :-)